### PR TITLE
build: robot builds for all release types

### PR DESCRIPTION
**Pull request by sfoster1**
Fix a couple build issues in the release branch:
- Some failures in the app yaml that were evident in the cold light of day, like incorrect glob patterns and bad step `if`s (this commit is in the release branch already unfortunately, you can see it here: https://github.com/Opentrons/opentrons/commit/59a588959c924567538bceb58bcfba00d0e3dec8 )
- robot builds actually need to be triggered on regular release branches and tags

### CODE REVIEW

**Review by vegano1 — APPROVED** [review]
makes sense.

**Review by b-cooper — APPROVED** [review]
⛓️ 🌲

File: scripts/deploy/create-release.js [hunk 1]
```diff
@@ -177,6 +177,15 @@ async function createRelease(token, tag, project, version, changelog, deploy) {
   }
 }
 
+function truncateAndAnnotate(changelog, limit, prevtag, thistag) {
+  if (changelog.length < limit) {
+    return changelog
+  }
+  const truncated = changelog.substring(0, limit)
+  const linkmessage = `...and more! Log link: https://github.com/${REPO_DETAILS.owner}/${REPO_DETAILS.repo}/compare/${prevtag}...${thistag}`
```
Comment (vegano1) [inline scripts/deploy/create-release.js]: Wouldn't this still fail here? since we are adding this message to the max message size dictated by `limit`?

Comment (sfoster1) [reply scripts/deploy/create-release.js]: yeah but I'm passing a value smaller than the actual limit to `limit`. i should probably document that

Comment (sfoster1) [reply scripts/deploy/create-release.js]: i made it take the length of the addon message into account specifically

Comment (vegano1) [reply scripts/deploy/create-release.js]: yep, that works.

**Pull request merged**